connector/src/main/java/org/jboss/as/connector/subsystems/datasources/Constants.java

```diff
@@ -492,6 +497,9 @@ public class Constants {
     static SimpleAttributeDefinition TRANSACTION_ISOLATION = new SimpleAttributeDefinitionBuilder(TRANSACTION_ISOLATION_NAME, ModelType.STRING, true)
             .setXmlName(DataSource.Tag.TRANSACTION_ISOLATION.getLocalName())
             .setAllowExpression(true)
+            .setAllowedValues(TransactionIsolation.TRANSACTION_NONE.name(), TransactionIsolation.TRANSACTION_READ_COMMITTED.name(),
+                    TransactionIsolation.TRANSACTION_READ_UNCOMMITTED.name(), TransactionIsolation.TRANSACTION_REPEATABLE_READ.name(),
+                    TransactionIsolation.TRANSACTION_SERIALIZABLE.name())
```

Comment:
This validation is incorrect per the existing code. See the uses of TransactionIsolation.customLevel(...) which allows arbitrary values.

Changing that would be a breaking change and the API would still have to allow the values on an HC for mixed-domain use, while rejecting them on a currently running server. Such a change should be handled under a different JIRA, assuming it's wanted.

Reply:
Ah, I didn't see the customLevel, we're losing the ability to show suggestions in CLI/HAL but the custom level makes that unfeasible anyway.
But the custom level would fail validation against the xsd, I'm assuming we're fine with that?

Reply:
We don't validate these user documents against the xsd, so in general, yes.

For the existing xsd versions, for sure yes.

For the new xsd I'll defer to @tadamski as to whether he wants to remove the restriction from the xsd.

If he wants to keep it in the xsd and this custom level thing is some sort of hidden thing, I suppose your setAllowedValues can be kept as well, since you reminded me it's just informational and doesn't trigger validation. If you do, please use the CLI to exercise the custom level thing; I don't remember clearly any more what if anything the CLI may do to try and help enforce allowed values.

Reply:
I gave a quick look at the history of why we needed a custom level since those allowed values are part of SQL standard, and it turns out that the culprit is MySQL's SNAPSHOT level.

MySQL seems like important use case, but since other values are standard based, I'm leaning towards keeping custom level hidden and keep setAllowedValues. Maybe it would also make sense to add a TransactionIsolation.SNAPSHOT (with comment that is SQLServer thing), and keep custom value hidden for some exotic cases.